src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/SchemaEvolutionService.java

```diff
@@ -12,5 +14,8 @@ public interface SchemaEvolutionService {
    *     nullability, and columns to add
    * @param record the sink record that contains the schema and actual data
    */
-  void evolveSchemaIfNeeded(SchemaEvolutionTargetItems targetItems, SinkRecord record);
+  void evolveSchemaIfNeeded(
+      SchemaEvolutionTargetItems targetItems,
+      SinkRecord record,
+      Map<String, ColumnProperties> schemaAlreadyInUse);
```

Comment:
wouldn't it be better name?
```suggestion
      Map<String, ColumnProperties> existingSchema);
```

Reply:
done